### ISSUE TITLE
Questions about test results and fine-tuning methods

### ISSUE THREAD

**Issue by 1st-hoan** [codeshop715/SPM#4]
Hello!
I am conducting an experiment inspired by your wonderful research.
However, the test results and fine-tuning settings differ from those in the paper.
The test results are as follows.
mini_imagenet : 5way-5shot
{"n_ways": 4.999975000125, "n_imgs": 99.99950000249999, **"acc1": 98.04684548583482**, "acc5": 99.99950000249999, "loss": 0.09170688536801204, "acc_std": 1.957693338394165, "epoch": -1}
mini_imagenet : 5way-1shot
{"n_ways": 4.999975000125, "n_imgs": 79.999600002, "**acc1": 92.83287171592096**, "acc5": 99.99950000249999, "loss": 0.2667618068488824, "acc_std": 6.445540904998779, "epoch": -1}
cifar_fs : 5way-5shot
{"n_ways": 4.999975000125, "n_imgs": 99.99950000249999, "**acc1": 92.50553999381492**, "acc5": 99.99950000249999, "loss": 0.22384315450882067, "acc_std": 5.9618754386901855, "epoch": -1}
cifar_fs : 5way-1shot
{"n_ways": 4.999975000125, "n_imgs": 79.999600002, "**acc1": 82.99958747398645**, "acc5": 99.99950000249999, "loss": 0.5060747203417434, "acc_std": 9.871161460876465, "epoch": -1}

In addition, the paper fine-tunes only about 2% (spatial prompt and channel prompt), but this is not applied in the code.
When I applied it myself, I got the following results. However, these also do not match the test results in the paper.
mini_imagenet : 5way-5shot
{"n_ways": 4.999975000125, "n_imgs": 99.99950000249999, "**acc1": 98.06551204518148**, "acc5": 99.99950000249999, "loss": 0.09405722724563097, "acc_std": 1.9566981792449951, "epoch": -1}
mini_imagenet : 5way-1shot
{"n_ways": 4.999975000125, "n_imgs": 79.999600002, "**acc1": 93.35086911258254**, "acc5": 99.99950000249999, "loss": 0.2578955240416788, "acc_std": 6.150838375091553, "epoch": -1}
cifar_fs : 5way-5shot
{"n_ways": 4.999975000125, "n_imgs": 79.999600002, "**acc1": 83.73425049340513**, "acc5": 99.99950000249999, "loss": 0.477284712814921, "acc_std": 9.51264476776123, "epoch": -1}
cifar_fs : 5way-1shot
{"n_ways": 4.999975000125, "n_imgs": 99.99950000249999, "**acc1": 92.63287271210129**, "acc5": 99.99950000249999, "loss": 0.2213962784277097, "acc_std": 5.7035698890686035, "epoch": -1}

If there are any missing settings in the provided code, please review them and let me know how to obtain the same results as in the paper.  
Thank you!

**Comment by MingC715** [comment]
Hi,

Thank you for your interest in our work. Your mini-ImageNet results actually look quite close to ours — the small differences could be due to random seeds.

For the remaining gaps (especially on CIFAR-FS), I'd suggest:
1. Tuning the learning rate during fine-tuning — i remember it has the biggest impact. 
2. Running with multiple seeds and averaging the results.

Hope this helps!
